CanonicalLayouts/supporting-pane-compose/app/src/main/java/com/example/supportingpanecompose/ui/SupportingPaneSample.kt

```diff
@@ -47,53 +54,82 @@ private val data = mapOf(
     "flutter" to listOf("android", "desktop")
 )
 
+@ExperimentalMaterial3AdaptiveApi
 @Composable
-fun SupportingPaneSample(
-    windowSizeClass: WindowSizeClass,
-    displayFeatures: List<DisplayFeature>
-) {
+fun SupportingPaneSample() {
     var selectedTopic: String by rememberSaveable { mutableStateOf(data.keys.first()) }
+    val navigator = rememberSupportingPaneScaffoldNavigator()
 
-    SupportingPane(
-        main = {
-            Column(
-                horizontalAlignment = Alignment.CenterHorizontally,
-                modifier = Modifier.fillMaxSize()
+    SupportingPaneScaffold(
+        supportingPane = {
+            AnimatedPane(
+                modifier = Modifier
+                    .background(
+                        color = Color.LightGray.copy(alpha = .3f),
+                        shape = RoundedCornerShape(16.dp)
+                    )
+                    .padding(all = 16.dp)
             ) {
-                Text("Main Content", style = MaterialTheme.typography.titleLarge)
-                Text(selectedTopic)
-            }
-        },
-        supporting = {
-            Column(
-                modifier = Modifier.fillMaxSize()
-            ) {
-                Text("Related Content", style = MaterialTheme.typography.titleLarge)
+                Column {
+                    Text(
+                        "Related Content",
+                        modifier = Modifier.padding(vertical = 16.dp),
+                        style = MaterialTheme.typography.titleLarge
+                    )
 
-                LazyColumn {
-                    items(
-                        data.getValue(selectedTopic),
-                        key = { it }
-                    ) { relatedTopic ->
-                        Box(
-                            Modifier
-                                .fillMaxWidth()
-                                .clickable {
-                                    selectedTopic = relatedTopic
-                                }
-                        ) {
-                            Text(
-                                text = relatedTopic,
-                                modifier = Modifier
-                                    .padding(16.dp)
-
-                            )
+                    LazyColumn {
+                        items(
+                            data.getValue(selectedTopic),
+                            key = { it }
+                        ) { relatedTopic ->
+                            Box(
+                                Modifier
+                                    .fillMaxWidth()
+                                    .padding(all = 4.dp)
+                                    .clickable {
+                                        selectedTopic = relatedTopic
+                                        if (navigator.canNavigateBack()) {
+                                            navigator.navigateBack()
```

Comment:
Is this because the default navigator is history aware so swapping supporting panes without this builds a backstack within the navigator?

Reply:
Without adding this, on `COMPACT` screen, when support pane is open, pressing back would get out of the `Scaffold` entirely